src/views/Farms/components/DepositModal.tsx

```diff
@@ -50,7 +52,12 @@ const DepositModal: React.FC<DepositModalProps> = ({ max, onConfirm, onDismiss,
         </Button>
         <Button
           width="100%"
-          disabled={pendingTx || fullBalance === '0' || val === '0'}
+          disabled={
+            pendingTx ||
+            parseFloat(fullBalance) === 0 ||
```

Comment:
Why don't you create 2 variables `BigNumber` instead of mixing parseFloat, Number, and BigNumber ?

Reply:
Done